pkg/compact/compact.go

```diff
@@ -470,6 +474,87 @@ func (cg *Group) Resolution() int64 {
 	return cg.resolution
 }
 
+// should return the results/metrics of the planning simulation
+type PlanSim interface {
+	PlanProgressCalc(ctx context.Context) error
+}
+
+type DefaultPlanSim struct {
+	grouper               Grouper
+	planner               Planner
+	sy                    *Syncer
+	numberOfIterations    prometheus.Gauge
+	numberOfBlocksToMerge prometheus.Gauge
+}
+
+func NewDefaultPlanSim(grouper Grouper, planner Planner, sy *Syncer, reg prometheus.Registerer) *DefaultPlanSim {
+	return &DefaultPlanSim{
+		grouper: grouper,
+		planner: planner,
+		sy:      sy,
+		numberOfIterations: promauto.With(reg).NewGauge(prometheus.GaugeOpts{
+			Name: "thanos_number_of_compaction_iterations",
+			Help: "The number of compactions to do",
+		}),
+		numberOfBlocksToMerge: promauto.With(reg).NewGauge(prometheus.GaugeOpts{
+			Name: "thanos_number_of_blocks_to_merge",
+			Help: "The number of blocks to be merged",
+		}),
+	}
+}
+
+func (ps *DefaultPlanSim) PlanProgressCalc(ctx context.Context) error {
```

Comment:
Let's figure out the signature here. 
I guess we can remove `syncer` and `grouper` from `DefaultPlanSim` and move it to the repeat loop in the main function.

For this signature, what about `func (ps *DefaultPlanSim) PlanProgressCalc(ctx context.Context, groups []*Group) error`?

This reduces the dependency of the PlanSimulator itself and should be easier for tests.

Reply:
It is correct. Nice work 👍